src/Microsoft.ML.TensorFlow/TensorflowTransform.cs

```diff
@@ -157,15 +256,299 @@ private static TensorFlowTransform Create(IHostEnvironment env, ModelLoadContext
         }
 
         // Factory method for SignatureDataTransform.
-        private static IDataTransform Create(IHostEnvironment env, Arguments args, IDataView input)
+        public static IDataTransform Create(IHostEnvironment env, Arguments args, IDataView input)
         {
             Contracts.CheckValue(env, nameof(env));
             env.CheckValue(args, nameof(args));
             env.CheckValue(input, nameof(input));
             env.CheckValue(args.InputColumns, nameof(args.InputColumns));
             env.CheckValue(args.OutputColumns, nameof(args.OutputColumns));
 
-            return new TensorFlowTransform(env, TensorFlowUtils.GetSession(env, args.Model), args.InputColumns, args.OutputColumns, TensorFlowUtils.IsSavedModel(env, args.Model) ? args.Model : null, false).MakeDataTransform(input);
+            return new TensorFlowTransform(env, args, input).MakeDataTransform(input);
+        }
+
+        internal TensorFlowTransform(IHostEnvironment env, Arguments args, IDataView input)
+            : this(env, TensorFlowUtils.GetSession(env, args.Model), args.InputColumns, args.OutputColumns, TensorFlowUtils.IsSavedModel(env, args.Model) ? args.Model : null, false)
+        {
+
+            Contracts.CheckValue(env, nameof(env));
+            env.CheckValue(args, nameof(args));
+
+            if (args.ReTrain)
+            {
+                // Shuffling fails because ShuffleTransform is not RowToRowMapper
+                // https://github.com/dotnet/machinelearning/issues/1106
+                //if(args.Shuffle)
+                //{
+                //    input = new ShuffleTransform(env, new ShuffleTransform.Arguments(), input);
+                //}
+
+                env.CheckValue(input, nameof(input));
+                env.CheckNonEmpty(args.OptimizationOperation, nameof(args.OptimizationOperation));
+
+                CheckParameters(args);
+
+                if (TensorFlowUtils.IsSavedModel(env, args.Model))
+                    TrainCore(args, args.Model, input);
+                else
```

Comment:
>else [](start = 16, length = 4)

Test for the opposite condition:
if (!TensorFlowUtils.IsSavedModel(...))
  throw ...
TrainCore(...);